app/controllers/reviews_controller.rb

```diff
@@ -63,43 +59,41 @@ def update
   # DELETE /reviews/1.json
   def destroy
     @restaurant = Restaurant.find(params[:restaurant_id])
-    @review = @restaurant.reviews.find(params[:id]);
+    @review = @restaurant.reviews.find(params[:id])
     @review.destroy
     respond_to do |format|
       format.html { redirect_to @restaurant, notice: 'Review was successfully destroyed.' }
       format.json { head :no_content }
     end
   end
 
-  private
+  def calc_rating(restaurant_id)
 
-    def set_restaurant
-    @restaurant = Restaurant.find(params[:restaurant_id])
-    end
+    @restaurant = Restaurant.find(restaurant_id)
 
-  # Use callbacks to share common setup or constraints between actions.
-  def set_review
-    @restaurant = Restaurant.find(params[:restaurant_id])
-    @review = @restaurant.reviews.find(params[:id])
+  unless @restaurant.reviews.empty?
+    @restaurant.rating = @restaurant.reviews.average(:rating)
   end
 
-    # Never trust parameters from the scary internet, only allow the white list through.
-    def review_params
-      params.require(:review).permit(:name, :rating, :comment)
-    end
+  @restaurant.save!
+end
 
+private
 
-  def calc_rating(curr_review)
+  def set_restaurant
+    @restaurant = Restaurant.find(params[:restaurant_id])
+  end
 
+  # Use callbacks to share common setup or constraints between actions.
+  def set_review
     @restaurant = Restaurant.find(params[:restaurant_id])
```

Comment:
you can remove the set_restaurant before filter and have only this one (for all methods but new i guess), something like this:
```
def initialize_params
   @restaurant = Restaurant.find(params[:restaurant_id]) if params[:restaurant_id].present?
   @review = Review.find(params[:id])
end
```